### PR TITLE
add auto-approve and tests

### PR DESCRIPTION
**Pull request by Abnormaltype**

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+from app import main
+
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+def test_should_be_declared():
+    assert (
+        hasattr(main, "calculate_profit") is True
+    ), "Function 'calculate_profit' should be declared."
```
Comment (danylott) [inline tests/test_main.py:16]: Unnecessary test.
You have import `calculate_profit` from app.main higher. So if it is not declared - the error will be on line 6

File: tests/test_main.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+def test_float_should_not_use():
+    code = inspect.getsource(calculate_profit)
+    parsed_code = ast.parse(code)
+    assert "float" not in ast.dump(parsed_code), (
+        "You should not use 'float()' inside 'calculate_profit', "
+        "because accuracy will be lost."
+    )
+
+
+def test_what_function_should_use():
```
Comment (danylott) [inline tests/test_main.py:34]:
```suggestion
def test_which_functions_should_be_used():
```

File: tests/test_main.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    ), "Function 'calculate_profit' should be declared."
+
+
+def test_function_should_return_none():
+    assert (
+        calculate_profit(f"{BASE_DIR}/app/trades.json") is None
+    ), "Function 'calculate_profit' should return None"
+
+
+def test_float_should_not_use():
```
Comment (danylott) [inline tests/test_main.py:25]:
```suggestion
def test_should_not_use_float():
```

File: tests/test_main.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    ), "You should use 'open()' inside 'calculate_profit'"
+    assert "dump" in ast.dump(
+        parsed_code
+    ), "You should use 'dump()' inside 'calculate_profit'"
+    assert "load" in ast.dump(
+        parsed_code
+    ), "You should use 'load()' inside 'calculate_profit'"
+    assert "json" in ast.dump(
+        parsed_code
+    ), "You should use 'json' module inside 'calculate_profit'"
```
Comment (danylott) [inline tests/test_main.py:52]: Interesting test btw

File: tests/test_main.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def test_create_profit_file():
+    expected = '{\n  "earned_money": "49.8176904",\n  "matecoin_account": "0.00007"\n}'
+
+    with open(f"{BASE_DIR}/profit.json") as actual:
+        assert actual.read() == expected
```
Comment (danylott) [inline tests/test_main.py:59]: This test is dependent on previous.
It's extremely bad.
Each test must be isolated.
So you need to call `calculate_profit` here for 100%

Comment (danylott) [reply tests/test_main.py:59]: Also, these tests does not cover any other cases, than in the example.
You need to create more different tests (mock `json.load` - and return different trades, than in example), and check, that result is ok

File: tests/test_main.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+def test_should_be_declared():
+    assert (
+        hasattr(main, "calculate_profit") is True
+    ), "Function 'calculate_profit' should be declared."
+
+
+def test_function_should_return_none():
+    assert (
+        calculate_profit(f"{BASE_DIR}/app/trades.json") is None
+    ), "Function 'calculate_profit' should return None"
```
Comment (danylott) [inline tests/test_main.py:22]: Your code doesn't clean up files after itself

**Review by danylott — CHANGES_REQUESTED** [review]

File: tests/test_main.py (hunk 5 above)
Comment (Abnormaltype) [reply tests/test_main.py]: fixed

**Review by danylott — APPROVED** [review]
LGTM

**Pull request merged**